src/main/java/io/jenkins/plugins/gitlabbranchsource/MergeRequestSCMHead.java

```diff
@@ -30,12 +31,13 @@
      * @param originName the name of the branch in the origin project
      * @param title the title of the merge request
      */
-    public MergeRequestSCMHead(@NonNull String name, long id, BranchSCMHead target,
+    public MergeRequestSCMHead(@NonNull String name, long id, BranchSCMHead target, String targetSha,
```

Comment:
Sorry to be blunt but this is plain and simple Wrong. The head must be revision agnostic or build storms will result

Reply:
Yes this is not right but why does `MergeRequestSCMRevision` not provide the correct target hash even when the correct target hash is provided during branch indexing. Any idea on that?

Reply:
Are you sure the revision is created correctly? The plugin is responsible for instantiating the revision with the required information. All the other plugins have this round-trip... so my suspicions would lie with the call to `new MergeRequestSCMRevision(...)` not having the correct arguments

Reply:
When I look at https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/master/src/main/java/io/jenkins/plugins/gitlabbranchsource/MergeRequestSCMRevision.java it seems to be storing a lot of extra stuff than say https://github.com/jenkinsci/gitea-plugin/blob/master/src/main/java/org/jenkinsci/plugin/gitea/PullRequestSCMRevision.java (which is also a Git based implementation). I wonder if conflicting information is being provided in your case

Reply:
Are these the correct way around? https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/835136aa930f798b795d2c6d0b123c605bb94b5c/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java#L272-L282

Reply:
Target should be the revision you are going to merge origin onto

Reply:
IOW for this PR, right now the target would be https://github.com/jenkinsci/gitlab-branch-source-plugin/commit/835136aa930f798b795d2c6d0b123c605bb94b5c and the origin would be https://github.com/jenkinsci/gitlab-branch-source-plugin/commit/9fa1f20e9996dd56ac36d8bead3279385d6bc12a

Reply:
Isn't ` mr.getDiffRefs().getBaseSha() ` the commit of the target branch that the merge request was created from, i.e. would only change if you rebased and force-pushed the merge request?

Reply:
Looking at the GitLab API I see three refs for the MR but no definition of what those refs actually are:

```
"diff_refs": {
    "base_sha": "c380d3acebd181f13629a25d2e2acca46ffe1e00",
    "head_sha": "2be7ddb704c7b6b83732fdd5b9f09d5a397b5f8f",
    "start_sha": "c380d3acebd181f13629a25d2e2acca46ffe1e00"
  },
```

I wonder if the definitions are somewhere else... perhaps `start_sha` is the current tip of the target branch and `base_sha` is the revision of the target branch that the MR was based off.. but  🤷‍♂ from the API docs that I have looked at

Reply:
Yes, your interpretation of the diffRefs definition is correct.

Reply:
> When I look at https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/master/src/main/java/io/jenkins/plugins/gitlabbranchsource/MergeRequestSCMRevision.java it seems to be storing a lot of extra stuff

The extra parameters are for the `getTrustedRevision` method that isn't implemented in the Gitea Plugin. See https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/master/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java#L663-L665. I think MRSCMRevision instantiation is correct in the plugin. 

As I said earlier I am printing the object with `toString` method defined inside MRSCMRevision, it prints the correct target hash which was used to instantiate it. But when baseHash (aka target hash) is passed inside `GitLabSCMBuilder` here https://github.com/jenkinsci/gitlab-branch-source-plugin/blob/master/src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMBuilder.java#L256-L261, it always supplies the incorrect the hash (i.e.  the commit of the target branch that the merge request was created from)

Reply:
> The extra parameters are for the getTrustedRevision method that isn't implemented in the Gitea Plugin.

It is implemented, but I'm not sure if @daniel-beck pushed the changes.

In any case you do not need to store that information as you have the `getTarget().getHash()` for the trusted revision. You are only double storing or risking confusion